test/unit/ngsiv2/provisioning/device-provisioning-api_test.js

```diff
@@ -0,0 +1,791 @@
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .post('/v2/entities?options=upsert',
+                    utils.readExampleFile('./test/unit/ngsiv2/examples/' +
+                        'contextRequests/createMinimumProvisionedDevice.json'))
+                .reply(204);
+
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert',
+                    utils.readExampleFile('./test/unit/ngsiv2/examples/' +
+                        'contextRequests/createMinimumProvisionedDevice.json'))
+                .reply(204);
+
+            done();
+        });
+
+        it('should accept both creations', function(done) {
+            request(options1, function(error, response, body) {
+                response.statusCode.should.equal(201);
+
+                request(options2, function(error, response, body) {
+                    response.statusCode.should.equal(201);
+                    done();
+                });
+            });
+        });
+
+        it('should show the new device in each list', function(done) {
+            request(options1, function(error, response, body) {
+                request(options2, function(error, response, body) {
+                    iotAgentLib.listDevices('smartGondor', '/gardens', function(error, results) {
+                        results.devices.length.should.equal(1);
+                        results.devices[0].id.should.equal('MicroLight1');
+
+                        iotAgentLib.listDevices('smartMordor', '/electricity', function(error, results) {
+                            results.devices.length.should.equal(1);
+                            results.devices[0].id.should.equal('MicroLight1');
+                            done();
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    describe('When there is a connection error with a String code connecting the CB', function() {
+        var options = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionMinimumDevice.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5